amazon/ion/equivalence.py

```diff
@@ -26,7 +26,7 @@
 from amazon.ion.symbols import SymbolToken
 
 
-def isinstance_of_ion_nature(obj):
+def isinstance_of_ion_nature_or_new_ion_py_objects(obj):
```

Comment:
This method will be deprecated with other old methods later.

Reply:
Right... But what will replace it? I ask because _I think_ that would be either just trying to get the `iontype` and `annotations` or testing if the two objects `hasattr` them. So why not do  that now and avoid the series of `or`s?

Reply:
Got it, refactored it to check if both ion_type and ion_annotations attributes exist.